lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb

```diff
@@ -2,101 +2,98 @@
+      @product = @cs_eti_table_products.product(name: @name)
     end
 
     context 'когда цена точная', story: 'когда цена точная' do
-      before do
-        @price = Faker::Number.number(2)
-        @cs_eti_page.set_price(@price)
-        @cs_eti_page.wait_saving
-        @cs_eti_page.search_product(@name)
-      end
+      let(:price) { {type: :exact, price: Faker::Number.number(2)} }
 
       it 'введеная цена отображается' do
-        expect(@cs_eti_page.product_price(@name)).to eq @price + ' руб.'
+        expect(@cs_eti_table_products.price(@product)).to eq(price[:price] + ' руб.')
       end
     end
 
     context 'когда цена от и до', story: 'когда цена от и до' do
       context 'когда только "от"' do
-        before do
-          @price_from = {from: Faker::Number.number(2)}
-          @cs_eti_page.set_price_from_to(@price_from)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
-        end
+        let(:price) { {type: :range, price: Faker::Number.number(2)} }
 
         it 'введеная цена отображается' do
-          expect(@cs_eti_page.product_price(@name)).to eq 'от ' + @price_from[:from] + ' руб.'
+          expect(@cs_eti_table_products.price(@product)).to eq 'от ' + price[:price] + ' руб.'
         end
       end
 
       context 'когда только "до"' do
-        before do
-          @price_to = {to: Faker::Number.number(2)}
-          @cs_eti_page.set_price_from_to(@price_to)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
-        end
+        let(:price) { {type: :range, price_max: Faker::Number.number(2)} }
 
         it 'введеная цена отображается' do
-          expect(@cs_eti_page.product_price(@name)).to eq 'до ' + @price_to[:to] + ' руб.'
+          expect(@cs_eti_table_products.price(@product)).to eq 'до ' + price[:price_max] + ' руб.'
         end
       end
 
       context 'когда заполняем "от" и "до"' do
-        before do
-          @price_from_to = {from: Faker::Number.number(2), to: Faker::Number.number(3)}
-          @cs_eti_page.set_price_from_to(@price_from_to)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
-        end
+        let(:price) { {type: :range, price: Faker::Number.number(2), price_max: Faker::Number.number(3)} }
 
         it 'введеная цена отображается' do
-          expect(@cs_eti_page.price_value).to include @price_from_to[:from], @price_from_to[:to]
+          expect(@cs_eti_table_products.price(@product)).to include price[:price], price[:price_max]
         end
       end
     end
 
     context 'когда цена со скидкой', story: 'когда цена со скидкой' do
-      before do
-        @discount_price = {previous: Faker::Number.number(3), discount: Faker::Number.number(2)}
-        @cs_eti_page.set_discount_price(@discount_price)
-        @cs_eti_page.wait_saving
-        @cs_eti_page.search_product(@name)
+      let(:price) do
+        {
+            type: :discount,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.